packages/gatsby/src/services/initialize.ts

```diff
@@ -1,54 +1,52 @@
-/* @flow */
-
-const _ = require(`lodash`)
-const { slash } = require(`gatsby-core-utils`)
-const fs = require(`fs-extra`)
-const md5File = require(`md5-file/promise`)
-const crypto = require(`crypto`)
-const del = require(`del`)
-const path = require(`path`)
-const Promise = require(`bluebird`)
-const telemetry = require(`gatsby-telemetry`)
-
-const apiRunnerNode = require(`../utils/api-runner-node`)
+import _ from "lodash"
+import { slash } from "gatsby-core-utils"
+import fs from "fs-extra"
+import md5File from "md5-file/promise"
+import crypto from "crypto"
+import del from "del"
+import path from "path"
+import telemetry from "gatsby-telemetry"
+
+import apiRunnerNode from "../utils/api-runner-node"
 import { getBrowsersList } from "../utils/browserslist"
-import { createSchemaCustomization } from "../utils/create-schema-customization"
+import { Store, AnyAction } from "redux"
+import { preferDefault } from "../bootstrap/prefer-default"
+import * as WorkerPool from "../utils/worker/pool"
+import JestWorker from "jest-worker"
 import { startPluginRunner } from "../redux/plugin-runner"
-const { store, emitter } = require(`../redux`)
+import { loadPlugins } from "../bootstrap/load-plugins"
+import { store, emitter } from "../redux"
+import loadThemes from "../bootstrap/load-themes"
+import reporter from "gatsby-cli/lib/reporter"
+import { getConfigFile } from "../bootstrap/get-config-file"
+import { removeStaleJobs } from "../bootstrap/remove-stale-jobs"
+import { IPluginInfoOptions } from "../bootstrap/load-plugins/types"
 import { internalActions } from "../redux/actions"
-const { loadPlugins } = require(`./load-plugins`)
-const loadThemes = require(`./load-themes`)
-const reporter = require(`gatsby-cli/lib/reporter`)
-import { getConfigFile } from "./get-config-file"
-const tracer = require(`opentracing`).globalTracer()
-import { preferDefault } from "./prefer-default"
-import { removeStaleJobs } from "./remove-stale-jobs"
+import { IGatsbyState } from "../redux/types"
+import { IBuildContext } from "./types"
 
 // Show stack trace on unhandled promises.
-process.on(`unhandledRejection`, (reason, p) => {
-  reporter.panic(reason)
+process.on(`unhandledRejection`, (reason: unknown) => {
+  // https://github.com/DefinitelyTyped/DefinitelyTyped/issues/33636
+  reporter.panic((reason as Error) || `Unhandled rejection`)
```

Comment:
Probably `new Error('Unhandled rejection')` or something? You could add that there was no explicit message :shrug:

Reply:
`reporter.panic` accepts all manner or arguments. I don't know whether a string or an error is better here tbh.